core/trino-main/src/main/java/io/trino/operator/aggregation/DecimalAverageAggregation.java

```diff
@@ -161,27 +161,34 @@ public static void inputLongDecimal(Type type, LongDecimalWithOverflowAndLongSta
 
     private static void accumulateValueInState(Slice unscaledValue, LongDecimalWithOverflowAndLongState state)
     {
-        accumulateAndUpdateOverflow(unscaledValue, state);
-        state.setLong(state.getLong() + 1);
-    }
-
-    private static void initializeIfNeeded(LongDecimalWithOverflowAndLongState state)
-    {
-        if (state.getLongDecimal() == null) {
-            state.setLongDecimal(UnscaledDecimal128Arithmetic.unscaledDecimal());
+        Slice sum = state.getLongDecimal();
+        if (sum == null) {
```

Comment:
See the below comment about maybe having and using something like:
 ```
 Slice sum = state.getOrInitializeLongDecimal(unscaledValue);
 if (sum != unscaledValue) {
   state.addOverflow(UnscaledDecimal128Arithmetic.addWithOverflow(sum, unscaledValue, sum));
 }
 ```

Reply:
Disregard, this strategy is ok for short decimal where we know it was just heap allocated but extremely dangerous for long decimals where you might have a Slice pointing into the middle of the input block.